### PR TITLE
Bugfix/ prevent uninstall of WINE and other programs

### LINKED ISSUE #2238: Installing Mycroft may uninstall WINE without notice
A [user on the forums reported](https://community.mycroft.ai/t/there-should-be-a-warning-that-mycroft-will-uninstall-wine-and-other-software/6995/2) that by installing Mycroft, their WINE installation and all programs using WINE were uninstalled without warning. Have requested further details about the users system.

This appears to have happened at least once before on Kubuntu 16.04
https://community.mycroft.ai/t/finally-got-it-working/5580
In that instance it seemed that the `portaudio19-dev` package required `libjack0` and thus uninstalled `lib-jack-2-0`.

#### Expected behaviour:
- [ ] Able to install Mycroft without removing other software from the system.
- [ ] If a package must be removed from the system this should provide a warning and require user confirmation with the option to abort the installation process and return the system to its previous state.

### PR DESCRIPTION
**Pull request by krisgesling**
## Description
Checks for the existence of `lib-jack-2-0` on debian systems and if found provides option to abort the install. Resolves #2238 

## How to test
Install WINE then attempt to install Mycroft

## Contributor license agreement signed?
- [x] CLA (Whether you have signed a [CLA - Contributor Licensing Agreement](https://mycroft.ai/cla/)

### CODE REVIEW

File: dev_setup.sh [hunk 1]
```diff
@@ -115,6 +115,34 @@ function get_YN() {
...
+function os_is() {
+    [[ $(grep "^ID=" /etc/os-release | awk -F'=' '/^ID/ {print $2}' | sed 's/\"//g') == $1 ]]
+}
+
+function os_is_like() {
+    [[ $(grep "^ID_LIKE=" /etc/os-release | awk -F'=' '/^ID_LIKE/ {print $2}' | sed 's/\"//g') == $1 ]]
+}
+
+if os_is_like debian || os_is debian || os_is_like ubuntu || os_is ubuntu || os_is linuxmint; then
+    if dpkg -V lib-jack-2-0 ; then
```
Comment (forslund) [inline dev_setup.sh]: We might want to use `dpkg -V lib-jack-2-0 > /dev/null 2>&1` to silence the output from the check.

**Comment by forslund** [comment]
This seems good. Is there any more changes you're planning to do or should I go ahead and merge it?

**Comment by krisgesling** [comment]
I just haven't been able to boot into Mint to test it out, GNOME Boxes doesn't seem to like their ISO's

**Comment by krisgesling** [comment]
+1 for real world testing, turns out I had the package name wrong
 
Have squashed the commits for the sake of mycroft-core's history

This should be good to go!

**Comment by forslund** [comment]
Excellent. Merging

**Pull request merged**